static/src/javascripts/projects/common/modules/check-dispatcher.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+import config from 'lib/config';
+import { allEmailCanRun, listCanRun } from 'common/modules/email/run-checks';
+import emailArticle from 'common/modules/email/email-article';
+import clash from 'common/modules/experiments/ab-test-clash';
+import trackAdRender from 'commercial/modules/dfp/track-ad-render';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
+import { checks } from './check-mediator-checks';
+import { resolveCheck, waitForCheck } from './check-mediator';
+
+const someCheckPassed = (results): boolean => results.includes(true);
+
+const everyCheckPassed = (results): boolean => !results.includes(false);
+
+/**
+    Any check added to checksToDispatch should also 
+    be added to the array of checks in './check-mediator-checks'. 
+**/
+const checksToDispatch = {
+    isOutbrainDisabled(): boolean {
+        if (commercialFeatures.outbrain) {
+            return false;
+        }
+
+        return true;
+    },
+
+    isUserInContributionsAbTest(): boolean {
+        return clash.userIsInAClashingAbTest(clash.contributionsTests);
+    },
+
+    isUserNotInContributionsAbTest(): Promise<boolean> {
+        return waitForCheck('isUserInContributionsAbTest').then(
+            userInContributionsAbTest => !userInContributionsAbTest
+        );
+    },
+
+    isUserInEmailAbTest(): boolean {
+        return clash.userIsInAClashingAbTest(clash.emailTests);
+    },
+
+    emailCanRunPreCheck(): boolean {
+        return allEmailCanRun();
+    },
+
+    listCanRun(): boolean {
+        const listConfigs = emailArticle.getListConfigs();
+
+        return !!Object.keys(listConfigs).find(key =>
+            listCanRun(listConfigs[key])
+        );
+    },
+
+    emailInArticleOutbrainEnabled(): boolean {
+        return config.switches.emailInArticleOutbrain;
+    },
+
+    hasHighPriorityAdLoaded(): any {
```

Comment:
Just noticed the signature of this and `hasLowPriorityAdLoaded` are inconsistent with other asynchronous checks in here. It might be dangerous to sometimes return a promise and sometimes not, as the calling function can't rely on the return value being `thennable`. Is it possible to make these function always return a `Promise<boolean>`, for safety?